File: bfs/그림.py
#백준 1926
from collections import deque 

board = []
dx = [-1, 1, 0, 0]
dy = [0, 0, 1, -1]


n, m = map(int, input().split())
vis = [[0 for _ in range(m)] for _ in range(n)]

for i in range(n):
		board.append(list(map(int, input().split())))

count = 0
sizelist = [0]

for row in range(n):
	for col in range(m):
		if vis[row][col]:
			continue
		vis[row][col] = 1
		if board[row][col] != 1:
			continue
		q = deque()
		q.append((row,col))
		count += 1
		size = 1
		while q:
			cur = q.popleft()
			for i in range(4):
				x = cur[0] + dx[i]
				y = cur[1] + dy[i]
				if x<0 or x>=n or y<0 or y>=m or vis[x][y] or board[x][y]!= 1:
					continue
				vis[x][y] = 1
				q.append((x,y))
				size += 1
		sizelist.append(size)
print(count)
print(max(sizelist))

File: bfs/그림.java
//baekjoon 1926
package src;

import java.util.*;

public class Main {
    static int[] dx = {-1,1,0,0};
    static int[] dy = {0,0,1,-1};
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] board = new int[n][m];
        int[][] visit = new int[n][m];

        for(int i = 0; i<n;i++){
            for(int j = 0; j<m;j++){
                board[i][j] = sc.nextInt();
            }
        }
        int count = 0;
        List<Integer> sizeList = new ArrayList<>();
        sizeList.add(0);

        for(int r = 0; r < n; r++){
            for(int c = 0; c<m; c++){
                if(visit[r][c] == 1)
                    continue;
                visit[r][c] = 1;
                if(board[r][c]!= 1)
                    continue;
                Queue<int[]> q= new LinkedList<>();
                q.offer(new int[]{r,c});
                count++;
                int size = 1;
                while(!q.isEmpty()){
                    int[] cur = q.poll();
                    for(int i = 0; i<4; i++){
                        int x = cur[0] + dx[i];
                        int y = cur[1] + dy[i];
                        if(x<0 || x>=n || y<0 || y>=m)
                            continue;
                        if(visit[x][y] == 1 || board[x][y]!= 1)
                            continue;
                        visit[x][y] = 1;
                        q.offer(new int[]{x,y});
                        size++;
                    }
                }
                sizeList.add(size);
            }

        }
        System.out.println(count);
        System.out.println(Collections.max(sizeList));

    }
}